test/integration/helpers/Npm.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Appium.Net.Integration.Tests.helpers
+{
+    internal class Npm
+    {
+
+        public static string GetNpmPrefixPath()
+        {
+            string npmPath = GetNpmExecutablePath();
+            string npmPrefixPath = RunCommand(npmPath, "-g root");
+
+            return npmPrefixPath.Trim();
+        }
+
+        private static string RunCommand(string command, string arguments, int timeoutMilliseconds = 30000)
+        {
+            try
+            {
+                using (Process process = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = command,
+                        Arguments = arguments,
+                        RedirectStandardOutput = true,
+                        RedirectStandardError = true,
+                        UseShellExecute = false,
+                        CreateNoWindow = true,
+                    }
+                })
+                {
+                    process.Start();
+                    string output = process.StandardOutput.ReadToEnd();
+                    string errorOutput = process.StandardError.ReadToEnd();
+                    _ = process.WaitForExit(timeoutMilliseconds);
+
+                    int exitCode = process.ExitCode;
+                    if (exitCode == 1)
+                    {
+                        throw new NpmUnknownCommandException($"Command: `{arguments}` exited with code {exitCode}. Error: {output}");
+                    }
+                    else if (exitCode != 0)
+                    {
+                        throw new ApplicationException($"Command: `{command}` exited with code {exitCode}. Error: {errorOutput}");
+                    }
+
+                    return output;
+                }
+            }
+            catch (ApplicationException ex)
+            {
+                throw ex;
+            }
+
+            catch (Win32Exception)
+            {
+                throw;
+            }
+            catch (NpmUnknownCommandException)
+            {
+                throw;
+            }
+            catch (Exception)
+            {
```

Comment:
what I actually meant could be found at https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/creating-and-throwing-exceptions

paragraph:

> When an argument to a method causes an exception. In this case, the original exception should be caught and an [ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception) instance should be created. The original exception should be passed to the constructor of the [ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception) as the [InnerException](https://learn.microsoft.com/en-us/dotnet/api/system.exception.innerexception) parameter

Reply:
I believe the above article contains a decent tutorial on general exceptions design

Reply:
I removed the ones I didn't have anything special to do there.